src/test/ui/consts/issue-64506.rs

```diff
@@ -8,7 +6,7 @@ pub struct ChildStdin {
 #[derive(Copy, Clone)]
 enum AnonPipe {}
 
-const FOO: () = {
+const FOO: () = { //~ evaluation of constant value failed
```

Comment:
I think this means https://github.com/rust-lang/rust/issues/64506 is not tested any more. The error was in the field access which we don't reach any more.

The tests that are changed by this PR need auditing to ensure they still test what they are supposed to test.

Reply:
In particular I think all cases where a pass-tests becomes a fail-test are wrong, and need to be fixed. But others might be wrong, too, if they now fail in a different spot than before.

Reply:
Changed this to use a privately uninhabited type. Maybe one other concern is the 93161 test, but I think it's okay because it is still `run-pass`.

Reply:
That one just gets some warnings, so should be fine.